pandas/io/common.py

```diff
@@ -878,6 +888,46 @@ def __next__(self) -> str:
         return newline.lstrip("\n")
 
 
+# Wrapper that wraps a StringIO buffer and reads bytes from it
+# Created for compat with pyarrow read_csv
+class BytesIOWrapper(io.BytesIO):
```

Comment:
I don't know what pandas's policy is on public/private classes. Maybe name the class `_BytesIOWrapper`. @jreback

Reply:
this is fine, this entire module is de-facto private.

Reply:
and we already expose TextIOWrapper so this is similar